contract_api/registry.py

```diff
@@ -0,0 +1,335 @@
+import datetime
+from collections import defaultdict
+
+import web3
+from common.repository import NETWORKS
+from common.utils import Utils
+from eth_account.messages import defunct_hash_message
+from pymysql import MySQLError
+
+
+class Registry:
+    def __init__(self, obj_repo):
+        self.repo = obj_repo
+        self.obj_utils = Utils()
+
+    def _get_all_service(self):
+        try:
+            all_orgs_srvcs_raw = self.repo.execute("SELECT O.org_id, O.owner_address, S.service_id  FROM service S, "
+                                                   "organization O WHERE S.org_id = O.org_id AND S.is_curated = 1")
+            all_orgs_srvcs = {}
+            for rec in all_orgs_srvcs_raw:
+                if rec['org_id'] not in all_orgs_srvcs.keys():
+                    all_orgs_srvcs[rec['org_id']] = {'service_id': [],
+                                                     'owner_address': rec['owner_address']}
+                all_orgs_srvcs[rec['org_id']]['service_id'].append(
+                    rec['service_id'])
+            return all_orgs_srvcs
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _get_all_members(self):
+        try:
+            all_orgs_members_raw = self.repo.execute(
+                "SELECT org_id, member FROM members M")
+            all_orgs_members = defaultdict(list)
+            for rec in all_orgs_members_raw:
+                all_orgs_members[rec['org_id']].append(rec['member'])
+            print(all_orgs_members)
+            return all_orgs_members
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def get_all_org(self):
+        try:
+            all_orgs_srvcs = self._get_all_service()
+            all_orgs_members = self._get_all_members()
+            all_orgs_data = []
+            for rec in all_orgs_srvcs:
+                data = {"org_id": rec,
+                        "owner_address": all_orgs_srvcs[rec]['owner_address'],
+                        "service_id": all_orgs_srvcs[rec]['service_id'],
+                        "members": all_orgs_members.get(rec, [])}
+                all_orgs_data.append(data)
+            return all_orgs_data
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _prepare_subquery(self, s, q, fm):
+        try:
+            sub_qry = ""
+            if s == "all":
+                for rec in fm:
+                    sub_qry += fm[rec] + " LIKE '" + str(q) + "%' OR "
+                sub_qry = sub_qry[:-3]
+            else:
+                sub_qry += fm[s] + " LIKE '" + str(q) + "%' "
+            return sub_qry.replace("org_id", "M.org_id")
+        except Exception as err:
+            raise err
+
+    def _get_total_count(self, sub_qry):
+        try:
```

Comment:
What does this method do? Are we getting a count of all services?

Reply:
Yes. I have added the docstring to describe a method.  Will update for all methods soon.